design/crd-configstatus.md

```diff
@@ -0,0 +1,348 @@
+- Not deleted when running `kubectl delete -f metallb.yaml` or `helm uninstall metallb`
+- Requires manual cleanup: `kubectl delete configurationstatus config-status -n metallb-system`
+- Other components never fail if they cannot patch their condition
+- Must review stale conditions, are they removed?
+
+# FEEDBACK from Fede
+
+https://github.com/metallb/metallb/pull/2856#issuecomment-3388777537
+
+> I am not sure I like this approach, and this is not exactly what was drafted
+> in the related design proposal.
+
+We need to clarify the drafted proposal, questions on the other doc
+
+> What the proposal said was either to cut corners making one single global
+> configuration status managed by the controller (because the controller is
+> responsible of the majority of the configuration) OR a per speaker
+> configuration (which might make sense now with all the node selectors we have).
+
+This proposal does have "single global config status" AND each speaker adds each status.
+
+Q: how the "node selectors" are important, AFAIC if there speaker it posts its status,
+if speaker stops existing condition becomes stale and removed (need some double checking
+on that)
+
+> With one singleton status, we may have a gigantic status, fed by all the
+> controllers of all the speakers, becoming a lot of data to carry around and to
+> redistribute to clients. A per node / per speaker instance is more scalable
+> imo.
+
+Status size is fixed (3 condition for the controller +
+N times 3 conditions per speaker) and the user should only check the summary.
+A solution which would list per node status and the user must "grep" the not valid
+is less ideal
+
+Q: What is "redistribute to clients"?
+
+> Regarding having one condition per controller, what we are giving away here is
+> the knowledge of the real error. A configuration is the result of many things:
+> * the nodes
+> * the configuration bits
+> * possibly, the result of applying the frrk8s configuration
+
+Q: what is "real" error example?
+
+Q: do you mean by the configuration bits the configuration of  all `kubectl api-resources --api-group=metallb.io -o name`
+AFAIC The validation if overall config is ok or not, it means if there is any
+error somewhere or do you mean to actively check (e.g. vtysh show bgp config)?
+
+> By hooking more or less blindly in each controller (even some that are
+> unnecessary, he service controller and the bgp status ones are the first two I
+> noticed), we are giving away the knowledge that the lower levels have, so the
+> same error might float differently because a node event was processed before of
+> a config event or viceversa.
+
+'hooking blindly' is not "unnecessary", e.g if the poolreconciler loses rbac to
+list something, that error will be shown, in all other case the condition will be ok.
+It is for free to post condition.
+
+Q: Can we define the "giving away the lower levels have? and "float differently"?
```

Comment:
By reporting the errors from all the controllers, we are giving up control on what we expose (because we are exposing everything) instead of being deliberate on what's really a configuration error (vs for example an rbac error). 

I'd like this to observe the same pattern we have in the other status controllers.
One status object changed whenever it's needed, a signal to one single controller in charge of changing it.
This pattern is more solid than changing the status from multiple different places.

Reply:
I replied above. Again, the webhooks cannot be trusted to cover everything as they don't cover eventual consistency, and the state they observe might be incomplete